app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -435,7 +435,8 @@ class HomeFragment : Fragment() {
             totalSites = settings.topSitesMaxLimit,
             frecencyConfig = TopSitesFrecencyConfig(
                 FrecencyThresholdOption.SKIP_ONE_TIME_PAGES
-            ) { !Uri.parse(it.url).containsQueryParameters(settings.frecencyFilterQuery) },
+            ) { !Uri.parse(it.url).isOpaque &&
+                    !Uri.parse(it.url).containsQueryParameters(settings.frecencyFilterQuery) },
```

Comment:
alternatively, you could add the check to the `containsQueryParameters` extension function

> fun Uri.containsQueryParameters(searchParameters: String): Boolean {
    if (searchParameters.isBlank() || !this.isOpaque) {
        return false
    }